### PR TITLE
java-eventing-shopping-cart with codegen

### PR DESCRIPTION
**Pull request by johanandren**
Refs #277

### CODE REVIEW

**Comment by johanandren** [comment]
Just fixing docs to compile with these changes, let's revisit how good the docs are separately and I'll continue with testkit instead for now.

File: samples/java-eventing-shopping-cart/src/main/java/shopping/cart/view/ShoppingCartViewServiceView.java [hunk 1]
```diff
@@ -1,41 +1,27 @@
...
 import org.slf4j.LoggerFactory;
 import shopping.cart.domain.ShoppingCartDomain;
-import shopping.cart.view.ShoppingCartViewModel;
 
-// user view impl
-public class ShoppingCartView extends AbstractShoppingCartView {
+public class ShoppingCartViewServiceView extends AbstractShoppingCartViewServiceView {
 
-  private static Logger LOG = LoggerFactory.getLogger(ShoppingCartView.class);
+  private static Logger LOG = LoggerFactory.getLogger(ShoppingCartViewServiceView.class);
```
Comment (raboof) [inline samples/java-eventing-shopping-cart/src/main/java/shopping/cart/view/ShoppingCartViewServiceView.java:14]: Ouch... I guess our convention should be not to suffix `Service` to protobuf services? Anyway not directly relevant to this PR.

File: samples/java-eventing-shopping-cart/src/main/proto/cart/shopping_cart_topic.proto [hunk 2]
```diff
@@ -33,6 +33,9 @@ message TopicOperation {
 }
 
 service ShoppingCartTopicService {
+    option (akkaserverless.service) = {
+        type : SERVICE_TYPE_ACTION
```
Comment (raboof) [inline samples/java-eventing-shopping-cart/src/main/proto/cart/shopping_cart_topic.proto:37]: Do we have a protobuf style guide somewhere? I think I'd use 2-space indents and no space between `type :`, but I'm not sure if we decided on anything. Anyway perhaps not for this PR.

**Review by raboof — APPROVED** [review]

File: samples/java-eventing-shopping-cart/src/main/java/shopping/cart/view/ShoppingCartViewServiceView.java (hunk 1 above)
Comment (johanandren) [reply samples/java-eventing-shopping-cart/src/main/java/shopping/cart/view/ShoppingCartViewServiceView.java:14]: Yeah, I agree it's not great, some related discussions in #238

File: samples/java-eventing-shopping-cart/src/main/proto/cart/shopping_cart_topic.proto (hunk 2 above)
Comment (johanandren) [reply samples/java-eventing-shopping-cart/src/main/proto/cart/shopping_cart_topic.proto:37]: I guess we should (at some point) lint these like we do in the framework project.

**Pull request merged**